### PR TITLE
Implemented SerialForwardMask base for GUI

### LINKED ISSUE #259: Port SerialForwardMask.java
Need to port `SerialForwardMask.java` class from `org.neurovillage.gui`

**Would you like to work on the issue?**
GSoC issue

### PR DESCRIPTION
**Pull request by jddeep**
Fixes #259 
**Changes**: Implemented SerialForwardMask base for GUI

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

### CODE REVIEW

**Review by CloudyPadmal — APPROVED** [review]

File: app/src/main/java/io/neurolab/gui/SerialForwardMask.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+    public SerialForwardMask(Context context, SerialForwardTask fwdTask, DefaultFFTData fftData, Config config) {
+        this.context = context;
+        this.fwdTask = fwdTask;
+        this.config = config;
+        df.setRoundingMode(RoundingMode.CEILING);
+
+        this.fftData = fftData;
+        serialForwarder = new SerialForwarder();
+
```
Comment (suresh-gandhi) [inline app/src/main/java/io/neurolab/gui/SerialForwardMask.java:41]: remove an extra line.

Comment (suresh-gandhi) [inline app/src/main/java/io/neurolab/gui/SerialForwardMask.java:41]: here also

File: app/src/main/java/io/neurolab/gui/SerialForwardMask.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            serialForwarder.disconnect();
+            try {
+                Thread.sleep(300);
+            } catch (InterruptedException e1) {
+                e1.printStackTrace();
+            }
+        }
+        boolean connectionSuccessful = connect();
+
+
```
Comment (suresh-gandhi) [inline app/src/main/java/io/neurolab/gui/SerialForwardMask.java:69]: extra line again

File: app/src/main/java/io/neurolab/gui/SerialForwardMask.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+import io.neurolab.main.network.SerialForwarder;
+import io.neurolab.main.task.SerialForwardTask;
+import io.neurolab.model.Config;
+import io.neurolab.model.DefaultFFTData;
+
+public class SerialForwardMask {
+    private static final long serialVersionUID = 1L;
+    public static int serialWidth = 580;
+    public static int serialHeight = 700;
+    DecimalFormat df = new DecimalFormat("#.##");
```
Comment (suresh-gandhi) [inline app/src/main/java/io/neurolab/gui/SerialForwardMask.java:19]: change the generic package protected scoping to private

**Review by suresh-gandhi — CHANGES_REQUESTED** [review]

**Comment by suresh-gandhi** [comment]
@jddeep you can use the reformat code and rearrange code options offered by IntelliJ. I mean it can be really helpful rather than do it manually which leads to errors as you have seen some times.

**Pull request merged**